week-1-ts-ecosystem/w01e03-object-oriented-structure/src/controllers/product-types.ts

```diff
@@ -0,0 +1,5 @@
+export enum AuctionType {
+  buy = "Buy now",
+  auction = "Auction",
+  free = "For free",
+}
```

Comment:
Jest to jedyne "tak czyste" rozwiązanie jakie widziałem bez użycia Generyków 🥇 .

Bardzo ładnie zrobione.
Jedyne co bym sprawdził to to czy tutaj zwykła unia wartości prostych na zasadzie:
`type AuctionType = 'BUY_NOW' | 'AUCTION' | 'FOR_FREE'`
by nie wystarczyła.

Chyba że widzisz jakąś dodatkową wartość na ten moment (albo przyszły) z użycia `enum` ?